trade-finance-manager-api/src/v1/controllers/auth/auth-service.js

```diff
@@ -0,0 +1,80 @@
+const { getTfmRolesFromEntraGroups } = require('../../helpers/entra-group-to-tfm-role');
+const authProvider = require('./auth-provider');
+const userController = require('../user/user.controller');
+const utils = require('../../../utils/crypto.util');
+const { updateLastLoginAndResetSignInData } = require('../user/user.controller');
+
+const getExistingTfmUser = async (entraUser) => {
+  const claims = entraUser.idTokenClaims;
+
+  const emails = [...claims.verified_primary_email, ...claims.verified_secondary_email];
+
+  return await userController.findByEmails(emails);
```

Comment:
Should we add some checks here, what happens if `entraUser` is empty or doesn't have e.g `verified_primary_email`

Reply:
In current tests user always had verified_primary_email, but it is difficult to tell state of all users in AD.